File: proxy-pattern/java/src/HlsPlayer.java
package src;

// HlsPlayer播放器实现了视频接口
public class HlsPlayer implements VideoPlayer {
  String fileType = null;
  String filePath = null;

  public HlsPlayer(String fileType, String filePath) {
    this.fileType = fileType;
    this.filePath = filePath;
  }

  public void play() {
    System.out.println("HlsPlayer path:" + filePath);
    System.out.println("HlsPlayer::play()");
  }
}

File: adapter-pattern/python/src/HlsPlayer.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.MediaPlayer import MediaPlayer

class HlsPlayer(MediaPlayer):

    def __init__(self, file_type, file_path):
        MediaPlayer.__init__(self, file_type, file_path)

    def play(self, ):
        print('HlsPlayer file_type:' + self.file_type + ' | file_path:' +
              self.file_path)
        print('HlsPlayer::play()')
